x-pack/plugins/security_solution/public/entity_analytics/api/hooks/use_preview_risk_scores.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useQuery } from '@tanstack/react-query';
+import dateMath from '@kbn/datemath';
+import { fetchPreviewRiskScore } from '../api';
+import type { RiskScorePreviewRequestSchema } from '../../../../common/risk_engine/risk_score_preview/request_schema';
+
+export const useRiskScorePreview = ({ range, filter }: RiskScorePreviewRequestSchema) => {
+  return useQuery(['POST', 'FETCH_PREVIEW_RISK_SCORE', range, filter], async ({ signal }) => {
+    const params: RiskScorePreviewRequestSchema = {};
+
+    if (range) {
+      const startTime = dateMath.parse(range.start)?.utc().toISOString();
+      const endTime = dateMath
+        .parse(range.end, {
+          roundUp: true,
+        })
+        ?.utc()
+        .toISOString();
+
+      if (startTime && endTime) {
```

Comment:
Should we inform the user if their date isn't being applied? How would they discover that fact?

Reply:
bump

Reply:
My understanding is that search bar already handles the wrong input for data, so the user can see that, we do need this check because `dateMath.parse` can return undefined. But it shouldn't happen in our use case

Reply:
👍